backend/app/models/match.rb

```diff
@@ -1,5 +1,5 @@
 class Match < ApplicationRecord
     validates :user1_id, allow_blank: false, :numericality => { :greater_than_or_equal_to => 0 }
-    validates :user2_id, :numericality => { :greater_than_or_equal_to => 0 }
+    validates :user2_id, allow_blank: true, :numericality => { :greater_than_or_equal_to => 0 }
```

Comment:
Are we sure we don't want `allow_nil: true` instead/as well?

Reply:
Also seems like we can make `user1_id` and `user2_id` foreign key constraints; any reason why we might not?

Reply:
I can add `allow_nil: true`. It seemed like it was working even when `user2_id` wasn't provided but it doesn't hurt.

Reply:
By foreign key constraints are you referring to adding `has_many :users` to the match controller?

Reply:
Yeah something like that.

Reply:
I added the `allow_nil: true`. As far as the foreign key constraints, I double checked and they are already in place. When you try to create a match with a user id that does not exist, it violates the foreign key constraint and fails.